File: wirecast/shotnames.pl
#!/usr/bin/perl
use strict;
use CGI;

require Win32::OLE;

#print "getting wirecast...\n";
my $w = Win32::OLE->GetActiveObject("Wirecast.Application");

if ($w) {
    my $d = $w->DocumentByIndex(1);
    my $l = $d->LayerByName("Normal");
    my $scount = $l->shotcount();
    for(my $i = 1; $i <= $scount; $i++) {
        my $s = $d->ShotByShotID($l->ShotIDByIndex($i));
        if(1 < $i) {
            print "\n";
        }
        print $s->{'name'};
    }

}else{
    print "couldn't get wirecast";
}

File: wirecast/shotnames.js
//Return an object with shots in parent layer
//wirecast.layer.shotnames();
// try {
//     var win32ole = require('win32ole');
//     var wirecast = win32ole.client.Dispatch('Wirecast.Application');
//     var layer = wirecast.DocumentByIndex(1).LayerByName('Normal');
//     shotnames(wirecast.DocumentByIndex(1), layer);
// }catch(e){
//     console.log('**Error caught***\n' + e);
// }

exports.shotnames = function(document, layer){
//function shotnames(document, layer) {
    try {
        var shotcount = layer.ShotCount;
        var shots = {}
        for(var i = 1; i <= shotcount; i++) {
            shots[i] = document.ShotByShotID(layer.ShotIDByIndex(i)).Name;
            //console.log("shot " + i + ":", document.ShotByShotID(layer.ShotIDByIndex(i)).Name);
        }
    }catch(e){
        console.log('**Error caught***\n' + e);
    }
    console.log(shots);
    return shots;
}
